src/Platforms/DB2Platform.php

```diff
@@ -241,20 +244,50 @@ public function getIndexDeclarationSQL(Index $index): string
      */
     protected function _getCreateTableSQL(string $name, array $columns, array $options = []): array
     {
-        $indexes = [];
-        if (isset($options['indexes'])) {
-            $indexes = $options['indexes'];
+        $columnListSql = $this->getColumnDeclarationListSQL($columns);
+
+        if (isset($options['uniqueConstraints']) && ! empty($options['uniqueConstraints'])) {
+            foreach ($options['uniqueConstraints'] as $definition) {
+                $columnListSql .= ', ' . $this->getUniqueConstraintDeclarationSQL($definition);
+            }
         }
 
-        $options['indexes'] = [];
+        if (isset($options['primary']) && ! empty($options['primary'])) {
+            $columnListSql .= ', PRIMARY KEY(' . implode(', ', array_unique(array_values($options['primary']))) . ')';
+        }
+
+        $statement = match ($options['temporary'] ?? '') {
+            '' => 'CREATE TABLE ',
+            'created' => 'CREATE GLOBAL TEMPORARY TABLE ',
+            'declared' => 'DECLARE GLOBAL TEMPORARY TABLE ',
+            default => throw new InvalidArgumentException(sprintf(
+                'invalid temporary specification for table %s',
+                $name,
+            ))
+        };
+
+        $query = $statement . $name . ' (' . $columnListSql;
+        $check = $this->getCheckDeclarationSQL($columns);
+
+        if (! empty($check)) {
+            $query .= ', ' . $check;
+        }
 
-        $sqls = parent::_getCreateTableSQL($name, $columns, $options);
```

Comment:
Removing the call to the parent implementation seems to bloat this method big time. Can we somehow restore that call and make the diff smaller?